packages/backend/src/plugins/auth.ts

```diff
@@ -24,5 +28,28 @@ export default async function createPlugin({
   config,
   discovery,
 }: PluginEnvironment): Promise<Router> {
-  return await createRouter({ logger, config, database, discovery });
+  return await createRouter({
+    logger,
+    config,
+    database,
+    discovery,
+    providerFactories: {
+      ...defaultAuthProviderFactories,
+      google: createGoogleProvider({
+        signIn: {
+          // resolver: 'email',
+          resolver: async ({ profile: { email } }, ctx) => {
+            if (!email) {
+              throw new Error('No email associated with user account');
+            }
+            const id = email.split('@')[0];
```

Comment:
Based on the conversation further down in the diff, this should be a full userRef right? e.g. `user:default/${id}`? Or we need to add parsing elsewhere that assumes `user:default` if it's a simple string.

Reply:
The latter.

For ownership, we'll probably just always look at the full `ent` contents so to speak - all of the user's identities and group memberships as returned in the Backstage token. So this id will pretty much be ignored at least in terms of ownership - it'll at most be an indicator of some sort of what identity is the "primary" one for the user, for some definition of primary that makes sense for the org. And indeed when reading it, one should assume default `User:default`.